### PR TITLE
Render `background-image: url(…)`

### PR DESCRIPTION
**Pull request by SimonSapin**
… and support most `background-*` properties.

Still to do:

* `background-attachment`
* Gradients

### CODE REVIEW

File: components/layout_thread_2020/lib.rs [hunk 1]
```diff
@@ -553,13 +563,15 @@ impl LayoutThread {
     fn build_layout_context<'a>(
         &'a self,
         guards: StylesheetGuards<'a>,
+        script_initiated_layout: bool,
```
Comment (nox) [inline components/layout_thread_2020/lib.rs]: Maybe an enum, `LayoutRequestOrigin` or something?

File: components/layout_thread_2020/lib.rs [hunk 2]
```diff
@@ -248,10 +260,10 @@ impl LayoutThreadFactory for LayoutThread {
...
         load_webfonts_synchronously: bool,
         window_size: WindowSizeData,
-        _dump_display_list: bool,
-        _dump_display_list_json: bool,
-        _dump_style_tree: bool,
-        _dump_rule_tree: bool,
+        dump_display_list: bool,
+        dump_display_list_json: bool,
+        dump_style_tree: bool,
+        dump_rule_tree: bool,
```
Comment (nox) [inline components/layout_thread_2020/lib.rs]: Such an unethical method. 😂 No nit here though, we don't have time for that.

File: components/layout_2020/Cargo.toml [hunk 3]
```diff
@@ -25,6 +25,7 @@ ipc-channel = "0.12"
 libc = "0.2"
 msg = {path = "../msg"}
 net_traits = {path = "../net_traits"}
+once_cell = "1.2.0"
```
Comment (nox) [inline components/layout_2020/Cargo.toml]: We already use `mitochondria`, which has an `OnceCell`, I think.

File: components/layout_2020/display_list.rs [hunk 4]
```diff
@@ -426,7 +426,7 @@ impl<'a> BuilderForBoxFragment<'a> {
             clipping_area_origin: f32,
             clipping_area_size: f32,
             positioning_area_size: f32,
-        ) -> (f32, f32) {
+        ) -> (bool, f32, f32) {
```
Comment (nox) [inline components/layout_2020/display_list.rs]: Struct would be nice here.

**Review by nox — COMMENTED** [review]
Nice comments all around.

File: components/layout_thread_2020/lib.rs (hunk 1 above)
Comment (SimonSapin) [reply components/layout_thread_2020/lib.rs]: This is copied from `components/layout_thread`

File: components/layout_thread_2020/lib.rs (hunk 2 above)
Comment (SimonSapin) [reply components/layout_thread_2020/lib.rs]: Yeah it’s pretty bad, but also copied over.

File: components/layout_2020/Cargo.toml (hunk 3 above)
Comment (SimonSapin) [reply components/layout_2020/Cargo.toml]: Switched

File: components/layout_2020/display_list.rs (hunk 4 above)
Comment (SimonSapin) [reply components/layout_2020/display_list.rs]: Done

**Comment by SimonSapin** [comment]
@bors-servo retry

https://github.com/servo/servo/issues/25523 ?

```
  ▶ TIMEOUT [expected OK] /html/semantics/forms/the-button-element/button-submit-remove-children-jssubmit.html
  │ 
  │ 
  │ 
  └ 
  ▶ Unexpected subtest result in /html/semantics/forms/the-button-element/button-submit-remove-children-jssubmit.html:
  │ TIMEOUT [expected PASS] This test will pass if a form navigation successfully occurs when clicking a child element of a <button type=submit> element with a onclick event handler which removes the button's child and then calls form.submit().
  └   → Test timed out

**Pull request merged**